src/ml_board/backend/restful_api/restful_api_server.py

```diff
@@ -67,18 +155,89 @@ def add_config_to_experiment(self, grid_search_id: str, experiment_id: str, conf
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+             checkpoint_resource (CheckpointResource) : CheckpointResource type
+
+        :returns: Pickle file Stream response
+        """
         try:
             file_generator = self.data_access.get_checkpoint_resource(grid_search_id=grid_search_id,
                                                                       experiment_id=experiment_id,
                                                                       epoch=epoch,
                                                                       checkpoint_resource=checkpoint_resource)
-            response = StreamingResponse(file_generator, media_type="application/octet-stream")
+            response = StreamingResponse(
+                file_generator, media_type="application/octet-stream")
             return response
         except InvalidPathError as e:
             raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
                                 detail="Provided invalid parameters for checkpoint resource.") from e
 
+    def del_checkpoint_resource_exec(self, grid_search_id: str, experiment_id: str, epoch: str, checkpoint_resource: CheckpointResource):
+        """
+        ``HTTP DELETE`` Delete checkpoint resource pickle file 
+          given the epoch, experiment ID & grid search ID.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+             checkpoint_resource (CheckpointResource) : CheckpointResource type    
+        """
+        try:
+            self.data_access.del_checkpoint_resource_exec(grid_search_id=grid_search_id,
+                                                          experiment_id=experiment_id,
+                                                          epoch=epoch,
+                                                          checkpoint_resource=checkpoint_resource)
+
+        except InvalidPathError as e:
+            raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
+                                detail="Provided invalid parameters for checkpoint resource.") from e
+
+    def post_checkpoint_resource_exec(self, grid_search_id: str, experiment_id: str, epoch: str, checkpoint_resource: CheckpointResource, file: bytes = File(...)):
+        """
+        ``HTTP POST`` Add a checkpoint resource pickle file 
+          given the epoch, experiment ID & grid search ID.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+             checkpoint_resource (CheckpointResource) : CheckpointResource type
+             file (bytes): Pickle file to be added
+
+        :returns: Pickle file Stream response
+        """
+        try:
+            payload_pickle = base64.b64decode(file)
```

Comment:
It is a variable used in FastApi in terms of the documentation. https://fastapi.tiangolo.com/tutorial/request-files/
b64encode is a better standard while sending binary files rather than using ASCII. https://stackoverflow.com/questions/3538021/why-do-we-use-base64